Add Render Deployment option

jakemalachowski opened the pull request:
Hello. I'm an engineer at [Render](https://render.com/). We recently set up a one-click deployment button that makes it super simple for your users to get up and running in just a couple minutes. I was hoping you would be willing to add us as an option in your readme. 

I'm happy to address any questions or concerns!

arp242 commented:
Looks nice; I think the current text you added looks a bit too much like, ehm, an ad though 😅 I'll probably make something similar for Linode this week as I need to deploy another server, and might as well their template setup thingy-ma-bob.

How do users update to a newer version?

---

I gave it a spin, and I got: *"An error occurred while planning the sync."*

In the logs it says:

```
Jul 24 06:14:04 AM  R> Initializing Goat Counter database schema...
Jul 24 06:14:04 AM  DO
Jul 24 06:14:04 AM  psql:/etc/schema.pgsql:27: ERROR:  duplicate key value violates unique constraint "pg_type_typname_nsp_index"
Jul 24 06:14:04 AM  DETAIL:  Key (typname, typnamespace)=(sites_id_seq, 2200) already exists.
Jul 24 06:14:05 AM  R> Goat Counter database schema is already initialized.
Jul 24 06:14:05 AM  R> Creating initial user account.
Jul 24 06:14:06 AM  zdb.Migrate.Check: select version: pq: relation "version" does not exist
```

I can connect to the PostgreSQL and that table exists; I did another manual deploy and now it seems to work. The initial "Setting up PostgreSQL"-step took quite a long time btw.

---

In the "Health Check Path" you can fill in /status: https://goatcounter-kypb.onrender.com/status – not sure if you can fill that in by default somewhere?

---

Also, it's spelled "GoatCounter", rather than "Goat Counter" 😅

jakemalachowski commented:
There was a fix to the first deploy failing that hadn't been merged. Sorry about that! It should work as expected now.

> looks a bit too much like, ehm, an ad though 😅 

Ha, fair enough. Updated. Feel free to suggest any further changes. I hope you don't mind having the Render URL in there so people can see what Render is before creating an account and deploying.

> How do users update to a newer version?

The version is managed in the `render-examples/goatcounter` [Dockerfile](https://github.com/render-examples/goatcounter/blob/master/Dockerfile), where we can keep it up to date. Users can also fork the repo and build their own repo just as easily as they can ours.

I also updated the [render.yaml](https://github.com/render-examples/goatcounter/blob/master/render.yaml#L5) to use `/status` for the health check.

arp242 commented:
Can you give me an account for this? My trail expired and now I can't test it any more.

To be honest I got so many errors with this thing that I'm hesitant to add it, and I'd like to make sure it actually work well first 😅 There's still a few things in that script that don't seem quite right to me at a glance, but not sure as I can't test.

---

I also added a `goatcounter db schema-pgsql` in the upcoming 1.4 release just now btw, so you don't need to do `ADD https://raw.githubusercontent.com/zgoat/goatcounter/c47bde96ad1de204fbcf573e5ca3a507330c4279/db/schema.pgsql /db/schema.pgsql` hackety hack stuff.

arp242 commented:
I'll close this now due to lack of feedback; let me know if you want to continue with this and we can get things moving again.
